homeassistant/helpers/customize.py

```diff
@@ -1,25 +1,57 @@
 """A helper module for customization."""
 import collections
-from typing import Dict, List
+from typing import Any, Dict, List
 import fnmatch
+import voluptuous as vol
 
 from homeassistant.const import CONF_ENTITY_ID
 from homeassistant.core import HomeAssistant, split_entity_id
+import homeassistant.helpers.config_validation as cv
 
-_OVERWRITE_KEY = 'overwrite'
-_OVERWRITE_CACHE_KEY = 'overwrite_cache'
+_OVERWRITE_KEY_FORMAT = '{}.overwrite'
+_OVERWRITE_CACHE_KEY_FORMAT = '{}.overwrite_cache'
 
+_CUSTOMIZE_SCHEMA_ENTRY = vol.Schema({
+    vol.Required(CONF_ENTITY_ID): vol.All(
+        cv.ensure_list_csv, vol.Length(min=1), [vol.Schema(str)], [vol.Lower])
+})
 
-def set_customize(hass: HomeAssistant, customize: List[Dict]) -> None:
+
+def _convert_old_config(inp: Any) -> List:
+    if not isinstance(inp, dict):
+        return cv.ensure_list(inp)
+    if CONF_ENTITY_ID in inp:
+        return [inp]  # sigle entry
+    res = []
+
+    inp = vol.Schema({cv.match_all: dict})(inp)
+    for key, val in inp.items():
+        val = dict(val)
+        val[CONF_ENTITY_ID] = key
+        res.append(val)
+    return res
+
+
+def get_customize_schema(
```

Comment:
My suggestion, make the core schema, the default, a CONSTANT and for zwave wrap the core constant in another vol.All limiting keys after the core one accepted them all.

Will this work?

(I know little about zwave except that it is probably the component with the most issues & PRs, so this must be as cleanly removed from the core as possible)